apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/BlockIndexService.java

```diff
@@ -21,4 +22,8 @@
     List<Long> getBlockIdsAfter(int height, int limit);
 
     int hardDeleteAllBlockIndex();
+
+    TransactionalDataSource getDataSourceWithSharding(long blockId);
```

Comment:
I guess, that these methods should be moved to another Interface, e.g. ShardDatasourceDiscoveryService, ShardDbExplorer, etc

Reply:
added ShardDbExplorer interface;